File: 2265-partition-array-according-to-given-pivot/2265-partition-array-according-to-given-pivot.java
import java.util.*;

class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int[] arr = new int[nums.length];
        int index = 0;

        // First pass: Add elements smaller than pivot
        for (int num : nums) {
            if (num < pivot) {
                arr[index++] = num;
            }
        }

        // Second pass: Add elements equal to pivot
        for (int num : nums) {
            if (num == pivot) {
                arr[index++] = num;
            }
        }

        // Third pass: Add elements greater than pivot
        for (int num : nums) {
            if (num > pivot) {
                arr[index++] = num;
            }
        }

        return arr;
    }
}

File: 2265-partition-array-according-to-given-pivot/2265-partition-array-according-to-given-pivot.py
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:
        ans = [0] * len(nums)  
        idx = 0
        n = len(nums)

        for i in range(0, n):
            if nums[i] < pivot:
                ans[idx] = nums[i]
                idx += 1
        
        for i in range(0, n):
            if nums[i] == pivot:
                ans[idx] = nums[i]
                idx += 1
        
        for i in range(0, n):
            if nums[i] > pivot:
                ans[idx] = nums[i]
                idx += 1

        return ans
